src/cast-player.js

```diff
@@ -88,22 +90,29 @@ class CastPlayer extends BaseRemotePlayer {
   _ended: boolean = false;
   _playbackStarted: boolean = false;
   _reset: boolean = true;
-  _destroyed: boolean = false;
+  _destroyed: boolean = true;
   _isOnLiveEdge: boolean = false;
   _mediaInfoIntervalId: IntervalID;
   _adsController: CastAdsController;
   _adsManager: CastAdsManager;
 
   constructor(castConfig: CastConfigObject, remoteControl: RemoteControl) {
     super('CastPlayer', castConfig, remoteControl);
-    CastLoader.load()
-      .then(() => {
-        this._initializeCastApi();
-        this._initializeRemotePlayer();
-      })
-      .catch(e => {
-        CastPlayer._logger.error('Cast initialized error', e);
-      });
+    const loadPromise = new Promise((resolve, reject) => {
+      if (!CastPlayer._isAvailable) {
```

Comment:
this means if we have 2 players on the page we will skip the load for the second player is it correct?
the issue here related to change media not to 2 players.

Reply:
It doesn't matter if its 2 players or one player that destroyed and created again.
The point here is that since cast SDK availability event is fired once, the availability state should be saved for the whole session and for all players.